File: androidScramdinger/src/main/java/ru/brauer/scrumdinger/android/extensions/ThemeUtils.kt
package ru.brauer.scrumdinger.android.extensions

import androidx.compose.ui.graphics.Color
import androidx.compose.ui.graphics.colorspace.ColorSpaces
import ru.brauer.scrumdinger.models.ColorParams
import ru.brauer.scrumdinger.models.ColorSpacesEnum
import ru.brauer.scrumdinger.models.Theme

val Theme.color: Color get() = colorParams.color

val ColorParams.color: Color
    get() = Color(
        red = red,
        green = green,
        blue = blue,
        alpha = alpha,
        colorSpace = when (colorSpace) {
            ColorSpacesEnum.SRGB -> ColorSpaces.Srgb
            ColorSpacesEnum.LINEAR_SRGB -> ColorSpaces.LinearSrgb
            ColorSpacesEnum.DISPLAY_P3 -> ColorSpaces.DisplayP3
        }
    )

File: iosScrumdinger/iosScrumdinger/ThemeUtils.swift
//
//  ThemeUtils.swift
//  iosScrumdinger
//
//  Created by Игорь Пивоваров on 06.01.2024.
//  Copyright © 2024 orgName. All rights reserved.
//

import SwiftUI
import scrumdingerShared

extension ColorSpacesEnum {
    var colorSpare: Color.RGBColorSpace {
        return switch(self) {
        case .srgb: .sRGB
        case .linearSrgb: .sRGBLinear
        case .displayP3: .displayP3
        default: .sRGB
        }
    }
}

extension ColorParams {
    var color: Color {
        return Color(self.colorSpace.colorSpare, red: Double(self.red), green: Double(self.green), blue: Double(self.blue), opacity: Double(self.alpha))
    }
}

extension Theme {
    var color: Color {
        return self.colorParams.color
    }
}
